iothub/device/src/Transport/Amqp/AmqpUnit.cs

```diff
@@ -372,6 +373,10 @@ internal async Task DisableMessageReceivingLinkAsync(CancellationToken cancellat
             if (_eventReceiverLinkDisconnected != null)
             {
                 _messageReceivingLink.Closed -= _eventReceiverLinkDisconnected;
```

Comment:
does this messaging link require a null check as well?

Reply:
we already have null checks for _messageReceivingLink in code above.